src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -1623,26 +1668,47 @@ private bool HasStringAndIntValuedAttribute(EntityHandle token, AttributeDescrip
             }
         }
 
-        private static bool CrackObsoleteAttributeData(out ObsoleteAttributeData value, ref BlobReader sig)
+#nullable enable
+        /// <summary>
+        /// Gets the well-known optional named properties on ObsoleteAttribute, if present.
+        /// Both 'diagnosticId' and 'urlFormat' may be present, or only one, or neither.
+        /// </summary>
+        /// <remarks>
+        /// Failure to find any of these properties does not imply failure to decode the ObsoleteAttribute,
+        /// so we don't return a value indicating success or failure.
+        /// </remarks>
+        private static (string? diagnosticId, string? urlFormat) CrackObsoleteProperties(ref BlobReader sig, IAttributeNamedArgumentDecoder decoder)
```

Comment:
>CrackObsoleteProperties [](start = 65, length = 23)

CrackObsoletePropertiesOrThrow?

Reply:
Alternatively, consider moving ```try/catch``` into this method and do not discard what we were able to read before things blow up. I.e. if we got ```diagnosticId``` before an exception, return it rather than discard.

---
In reply to: [398713272](https://github.com/dotnet/roslyn/pull/42518#discussion_r398713272) [](ancestors = 398713272)

Reply:
Done